__tests__/ui/get-markup.test.js

```diff
@@ -1,214 +1,235 @@
+            };
+
+            const result = getMarkup(structuredPrice);
+            const header = result.querySelector('header');
+
+            expect(header.textContent).toContain(structuredPrice.name);
+            expect(header.textContent).toContain(priceDisplay(price).textContent);
+
+            const content = result.querySelector('.dim');
+
+            expect(content.children.length).toBe(2);
+            expect(content.firstChild).toBeInstanceOf(HTMLParagraphElement);
+            expect(content.firstChild.textContent).toBe(text.text);
+
+            expect(content.lastChild).toBeInstanceOf(HTMLAnchorElement);
+            expect(content.lastChild.href).toBe(link.url);
+            expect(content.lastChild.target).toBe('_blank');
+            expect(content.lastChild.rel).toBe('noopener');
+            expect(content.lastChild.textContent).toBe(link.name);
+        });
+    });
 
-        const result = getMarkup(Link);
-        expect(result).toEqual(expect.objectContaining({
-            values: expect.arrayContaining([
-                Link.name,
-                Link.url
-            ])
-        }));
+    describe('structured text', () => {
+        it('renders the structure and content to an article', () => {
+            const text = {
+                type: 'Text',
+                text: 'Copyrights© Limited'
+            };
+            const link = {
+                type: 'Link',
+                url: 'https://example.com/',
+                name: 'Link Name'
+            };
+            const structuredText = {
+                type: 'StructuredText',
+                name: 'Here\'s the legal bit',
+                contents: [text, link]
+            };
+
+            const result = getMarkup(structuredText);
+            const header = result.querySelector('header');
+
+            expect(header.textContent).toContain(structuredText.name);
+
+            const content = result.querySelector('.dim');
+
+            expect(content.children.length).toBe(2);
+            expect(content.firstChild).toBeInstanceOf(HTMLParagraphElement);
+            expect(content.firstChild.textContent).toBe(text.text);
+
+            expect(content.lastChild).toBeInstanceOf(HTMLAnchorElement);
+            expect(content.lastChild.href).toBe(link.url);
+            expect(content.lastChild.target).toBe('_blank');
+            expect(content.lastChild.rel).toBe('noopener');
+            expect(content.lastChild.textContent).toBe(link.name);
+        });
     });
 
-    describe('File', () => {
-        // TODO
+    describe('file', () => {
+        let fileDeferred;
+        let sdkMock;
+        let file;
+
+        beforeEach(() => {
+            fileDeferred = new Deferred();
+            sdkMock = {
+                getJobFile: jest.fn(() => fileDeferred.promise)
+            };
+            file = {
+                type: 'File',
+                filename: 'file-name.txt',
+                name: 'a-name',
```

Comment:
`name` was optional, `filename` mandatory, BUT `name` is preferable when available – not sure it changed in the Protocol and all scripts are updated already, so we better be prepared for `name` and no `name` scenarios

Reply:
Good point. I'll add tests to cover this.